### PR TITLE
Update to Avro 1.9.0 and remove patched Avro code

### PR DESCRIPTION
**Pull request by ajborley**

### CODE REVIEW

**Comment by massdosage** [comment]
At a quick glance these changes look good, thanks for the contribution! @teabot could you take a proper look please?

**Comment by ajborley** [comment]
@teabot - would be great to get a review on this. We would like to update our project to Avro 1.9.0 and would like to use an 'official' version of this package with that.

**Comment by massdosage** [comment]
@nahguam in @teabot's absence, any issues with merging this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.1.2] - 2019-08-09
```
Comment (massdosage) [inline CHANGELOG.md]: Let's go with 2.2.0 as the new version number as this is more than a bugfix. Can you please update here (you can leave the date as "TBD" and we'll fill it in when we do a release) and also in the pom.xml (change to 2.2.0-SNAPSHOT, we'll set the version when we do a release).

Comment (ajborley) [reply CHANGELOG.md]: @massdosage Thanks for taking a look. I've added a new commit with the version number changes.

**Pull request merged**